griptape/events/base_event.py

```diff
@@ -8,3 +12,47 @@ class BaseEvent(ABC):
     timestamp: float = field(
         default=Factory(lambda: time.time()), kw_only=True
     )
+    type: str = field(default=Factory(lambda self: self.__class__.__name__, takes_self=True), kw_only=True)
+
+    @classmethod
+    def from_dict(cls, event_dict: dict) -> BaseEvent:
```

Comment:
If the use wants to implement a new event will they have to override this method? There is a way to abstract this but I don't know if it makes sense at this stage.

For now, I think, we can just add a section in the docs describing how to add and use custom events.

Reply:
They would. I had copied the implementation from [BaseArtifact](https://github.com/griptape-ai/griptape/blob/main/griptape/artifacts/base_artifact.py#L34), do we need to update docs for that as well? What did you have in mind for generalizing it? The current implementation of using `class_registry` feels wrong as it is described as an [private API](https://marshmallow.readthedocs.io/en/stable/_modules/marshmallow/class_registry.html).

Reply:
I just realized that what I had in mind won't work. Probably not worth worrying about for now.